privcount/tally_server.py

```diff
@@ -398,11 +410,24 @@ def start_new_collection_phase(self, dc_uids, sk_uids):
         for uid in sk_uids:
             sk_public_keys[uid] = self.clients[uid]['public_key']
 
+        counters = self.config['counters']
+        model = None
+        if 'traffic_model' in self.config:
+            # load the model json file
+            inf = open(self.config['traffic_model'], 'r')
+            model = json.load(inf)
+            inf.close()
+
+            # add all of the counters needed to count the model states
+            for label in TrafficModel(model['states'], model['start_probability'], model['transition_probability'], model['emission_probability']).get_counter_labels():
+                # XXX TODO FIXME I think this way of initializing these new labels is fragile, wrong, or both
+                counters[label] = {'bins': [[0.0, float("inf")]]}
```

Comment:
It seems OK to me, but they are missing a sigma value, which might or might not matter depending on the context in which they are used.

What are your concerns in particular?